Divergic.Logging.Xunit/LoggingConfig.cs

```diff
@@ -7,21 +12,63 @@
     public class LoggingConfig
     {
         /// <summary>
-        ///     Initializes a new instance of the <see cref="LoggingConfig" /> class.
+        ///     Gets or sets whether exceptions thrown while logging outside of the test execution will be ignored.
         /// </summary>
-        public LoggingConfig()
-        {
-            Formatter = new DefaultFormatter();
-        }
+        public virtual bool IgnoreTestBoundaryException { get; set; } = false;
 
         /// <summary>
-        ///     Gets or sets a custom formatting for rendering log messages to xUnit test output.
+        ///     Identifies the number of spaces to use for indenting scopes.
         /// </summary>
-        public ILogFormatter Formatter { get; set; }
+        public int PaddingSpaces { get; set; } = 3;
 
         /// <summary>
-        ///     Gets or sets whether exceptions thrown while logging outside of the test execution will be ignored.
+        ///     Formats the log message with the specified values.
         /// </summary>
-        public bool IgnoreTestBoundaryException { get; set; }
+        /// <param name="scopeLevel">The number of active logging scopes.</param>
+        /// <param name="name">The logger name.</param>
+        /// <param name="logLevel">The log level.</param>
+        /// <param name="eventId">The event id.</param>
+        /// <param name="message">The log message.</param>
+        /// <param name="exception">The exception to be logged.</param>
+        /// <returns>The formatted log message.</returns>
+        public virtual string Format(
```

Comment:
This breaks Single Responsibility Pattern. Formatting is an action that in itself is not configuration. Please revert to how it was using an ILogFormatter property.

You could still combine these concepts with your own custom config class though. Perhaps something like this that also provides a simple singleton access property.

```
public class MyCustomConfig : LoggingConfig, ILogFormatter
{
    private MyCustomConfig()
    {
        Formatter = this;
    }

    public static MyCustomConfig Current { get; } = new MyCustomConfig();
    
    public string Format(
        int scopeLevel,
        string name,
        LogLevel logLevel,
        EventId eventId,
        string message,
        Exception exception)
    {
        var sb = new StringBuilder();

        if (scopeLevel > 0)
        {
            sb.Append(' ', scopeLevel * 2);
        }

        sb.Append($"{GetShortLogLevelString(logLevel)} ");

        if (!string.IsNullOrEmpty(name))
        {
            sb.Append($"{name} ");
        }

        if (eventId.Id != 0)
        {
            sb.Append($"[{eventId.Id}]: ");
        }

        if (!string.IsNullOrEmpty(message))
        {
            sb.Append(message);
        }

        if (exception != null)
        {
            sb.Append($"\n{exception}");
        }

        return sb.ToString();
    }

    private static string GetShortLogLevelString(LogLevel level)
    {
        switch (level)
        {
            case LogLevel.Trace: return "Trace";
            case LogLevel.Debug: return "Debug";
            case LogLevel.Information: return "Info ";
            case LogLevel.Warning: return "Warn ";
            case LogLevel.Error: return "Error";
            case LogLevel.Critical: return "Crit ";
            case LogLevel.None: return "None ";
            default: throw new Exception("invalid");
        }
    }
}
```

Reply:
"Formatting is an action that in itself is not configuration." True. What about changing the class name of the class from LoggingConfig to LoggingProcessor/Outputter/Adapter/Generator/Producer? Then you could claim the class has a single, although more general, responsibility. Honoring SRP would sacrifice simplicity, which I believe is more valuable.

Reply:
Sorry, been busy. I take your point however I like the scope of the config types to describe config as it is the most appropriate description for that purpose in the API. The workaround I provided still gives you the simplicity in the calling implementation.